Makefile

```diff
@@ -2,29 +2,37 @@ BUILDDIR=./build
 GOTIFY_VERSION=master
 PLUGIN_NAME=myplugin
 PLUGIN_ENTRY=plugin.go
+GO_VERSION=`cat $(BUILDDIR)/gotify-server-go-version`
+UPDATE_WITH_GIMME=true
 
 download-tools:
 	GO111MODULE=off go get -u github.com/gotify/plugin-api/cmd/gomod-cap
 
-create-bulid-dir:
+create-build-dir:
 	mkdir -p ${BUILDDIR} || true
 
-check-go-mod: create-bulid-dir
+update-go-mod: create-build-dir
 	wget -LO ${BUILDDIR}/gotify-server.mod https://raw.githubusercontent.com/gotify/server/${GOTIFY_VERSION}/go.mod
-	gomod-cap -from ${BUILDDIR}/gotify-server.mod -to go.mod -check=true
+	gomod-cap -from ${BUILDDIR}/gotify-server.mod -to go.mod
 	rm ${BUILDDIR}/gotify-server.mod || true
+	go mod tidy
 
-build-linux-amd64:
-	 docker run --rm -v "$$PWD/.:/proj" -w /proj gotify/build:v1-linux-amd64 go build -a -installsuffix cgo -ldflags "-w -s" -buildmode=plugin -o build/${PLUGIN_NAME}-linux-amd64.so /proj
+get-gotify-server-go-version: create-build-dir
+	rm ${BUILDDIR}/gotify-server-go-version || true
+	wget -LO ${BUILDDIR}/gotify-server-go-version https://raw.githubusercontent.com/gotify/server/${GOTIFY_VERSION}/GO_VERSION
 
-build-linux-arm-7:
-	 docker run --rm -v "$$PWD/.:/proj" -w /proj gotify/build:v1-linux-arm-7 go build -a -installsuffix cgo -ldflags "-w -s" -buildmode=plugin -o build/${PLUGIN_NAME}-linux-arm-7.so /proj
+change-go-version: get-gotify-server-go-version
+	${UPDATE_WITH_GIMME} && eval "$(shell gimme $(GO_VERSION))"
```

Comment:
As you said in https://github.com/gotify/plugin-template/issues/2#issuecomment-462112825, this won't apply to the building container, so I think has no effect on the build?

Reply:
The behaviour of the `go mod *`-commands could be different with different Go versions.
Tho' I'm also unsure if this is really needed.

Reply:
I think we do not need this. If the behavior changes, we just need to make sure that an equivlent of `go mod tidy` would be used, a `go mod tidy`-like feature is essential to the go module system and it would always be there. If `go mod tidy` vanished it must been a major breaking change to how go handles modules and we would need to modify much more than this line:(

Reply:
Oke.